server/src/main/java/io/crate/auth/JWTAuthenticationMethod.java

```diff
@@ -149,10 +156,14 @@ It adds hard coded WELL_KNOWN_JWKS_PATH (/.well-known/jwks.json) to the url.
         } catch (MalformedURLException | URISyntaxException e) {
             throw new IllegalArgumentException("Invalid jwks uri", e);
         }
-
-        return new JwkProviderBuilder(jwkUrl).build();
+        JwkProviderBuilder jwkProviderBuilder = new JwkProviderBuilder(jwkUrl);
+        if (ttl != null) {
+            jwkProviderBuilder.cached(5, ttl);
```

Comment:
Typically, it's only 1 key being returned but 5 is default afaik.
We can add a comment that this is default cache size per JWK endpoint.